sound/soc/sof/topology.c

```diff
@@ -1859,14 +1860,19 @@ static int sof_widget_load_pga(struct snd_soc_component *scomp, int index,
 	}
 
 	/* configure volume IPC message */
-	volume->comp.hdr.size = sizeof(*volume);
+	volume->comp.hdr.size = ipc_size;
 	volume->comp.hdr.cmd = SOF_IPC_GLB_TPLG_MSG | SOF_IPC_TPLG_COMP_NEW;
 	volume->comp.id = swidget->comp_id;
 	volume->comp.type = SOF_COMP_VOLUME;
 	volume->comp.pipeline_id = index;
 	volume->comp.core = core;
 	volume->config.hdr.size = sizeof(volume->config);
 
+	/* append extended data to the end of the component */
+	memcpy((u8 *)volume + sizeof(*volume),
```

Comment:
`volume + 1` and more similar cases later

Reply:
similar, if we want it, we need to change all of them together.